akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -119,7 +125,7 @@ private[io] object SelectionHandler {
 
     private[this] val select = new Task {
       def tryRun(): Unit = {
-        if (selector.select() > 0) { // This assumes select return value == selectedKeys.size
+        if (selector.select(MaxSelectMillis) > 0) { // This assumes select return value == selectedKeys.size
```

Comment:
Wait, in case of a timeout won't this return `0` and as such not do anything now? ("The number of keys, possibly zero, whose ready-operation sets were updated", https://docs.oracle.com/javase/7/docs/api/java/nio/channels/Selector.html#select())